port/unix/omrsysinfo.c

```diff
@@ -5884,7 +5884,12 @@ isRunningInContainer(struct OMRPortLibrary *portLibrary, BOOLEAN *inContainer)
 	/* Assume we are not in container */
 	*inContainer = FALSE;
 
-	if (isCgroupV1Available(portLibrary)) {
+	/* Check for existence of files that signify running in a container (Docker and Podman respectively). Not completely
+	 * reliable as these files may not be available, but it should work for most cases.
+	 */
+	if ((0 == access("/.dockerenv", F_OK)) || (0 == access("/run/.containerenv", F_OK))) {
+		*inContainer = TRUE;
+	} else if (isCgroupV1Available(portLibrary)) {
```

Comment:
The below check does not apply to cgroup v2. We should avoid reading `/proc/1/cgroup` for cgroup v2. Perf will be better. Is it guaranteed that `isCgroupV1Available` will always fail for cgroup v2?

Reply:
For cgroup v2 tmpfs will not be mounted on /sys/fs/cgroup, instead, 'cgroup' filesystem type is mounted there, so I believe our checks for v1 and v2 are mutually exclusive.

Reply:
Why isn't reading /proc/1/cgroup correct for newer v1? From what I have seen cgroup v1 dockers have /proc/1/cgroup outputting nonroot cgroups

Reply:
I wanted to refer to your old comment: https://github.com/eclipse/omr/pull/6422#issuecomment-1071249090 where you mentioned `we may need to change isRunningInContainer for cgroup v1 as well. I switched back to v1 ... All of the cgroups here are root.`

Reply:
Ok, I believe what's happening there is namespaces being enabled by the v2 container. Namespaces can also be enabled in v1 so I will add a comment explaining that